File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        
        slow = 1
        n = len(nums)
        count = 1

        
        for fast in range(1,n):
            if nums[fast] == nums[fast - 1]:
                count += 1 
                    
            else:
                count = 1 
                
            if count < 3:
                nums[slow] = nums[fast]
                slow += 1
        
        return slow
        

File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if(nums == null || nums.length == 0){
            return 0;
        }

        int count = 1;
        int slow = 1;
        int n = nums.length;

        for(int fast = 1; fast < n; fast++){
            if(nums[fast] == nums[fast - 1]){
                count += 1;
            }else{
                count = 1;
            }

            if(count < 3){
                nums[slow] = nums[fast];
                slow++;
            }
        }

        return slow;
    }
}


//two pointers
// # time complexity: O(n)
// # space complexity: O(1)